sirepo/server.py

```diff
@@ -540,17 +541,16 @@ def api_srUnit():
 def api_staticFile(path_info=None):
     """flask.send_from_directory for static folder.
 
-    Uses safe_join which doesn't allow indexing, paths with '..',
-    or absolute paths.
-
     Args:
         path_info (str): relative path to join
     Returns:
         flask.Response: flask.send_from_directory response
     """
     if not path_info:
-        raise sirepo.util.raise_not_found('empty path info')
-    p = pkio.py_path(flask.safe_join(str(simulation_db.STATIC_FOLDER), path_info))
+        sirepo.util.raise_not_found('empty path info')
+    # SECURITY: Use check_input to verify that the path does not escape the
+    # static dir.
+    p = sirepo.resource.static(path_info, check_input=True)
```

Comment:
Here's an example that explicitly calling safe_path here would be better. You wouldn't need to write this comment, instead it would look like:

`sirepo.resource.static(sirepo.util.safe_path(path_info))`

Much clearer to me.

Reply:
6e7eede4d